crate/utils/src/access.rs

```diff
@@ -91,16 +91,15 @@ impl<'de> Deserialize<'de> for ExtraDatabaseParams {
     where
         D: serde::Deserializer<'de>,
     {
-        let bytes = <Vec<u8>>::deserialize(deserializer)?;
+        let bytes = Zeroizing::from(<Vec<u8>>::deserialize(deserializer)?);
         let group_id_bytes: [u8; 16] = bytes[0..16]
             .try_into()
             .map_err(|_| serde::de::Error::custom("Could not deserialize ExtraDatabaseParams"))?;
         let group_id = u128::from_be_bytes(group_id_bytes);
         let key_bytes: [u8; 32] = bytes[16..48]
             .try_into()
             .map_err(|_| serde::de::Error::custom("Could not deserialize ExtraDatabaseParams"))?;
-        let key = SymmetricKey::try_from_bytes(key_bytes)
-            .map_err(|_| serde::de::Error::custom("Could not deserialize ExtraDatabaseParams"))?;
+        let key = Secret::from_protected_bytes(&key_bytes);
```

Comment:
Why not calling `from_unprotected_bytes` directly on the slice `&bytes[16..48]` ? You would avoid a copy.

Reply:
Because bytes are already zeroized.